buildingmotif/utils.py

```diff
@@ -692,3 +693,22 @@ def skolemize_shapes(g: Graph) -> Graph:
     # apply the replacements
     replace_nodes(g, replacements)
     return g
+
+
+def approximate_graph_hash(graph: Graph) -> int:
+    """
+    Returns a cryptographic hash of the graph contents.
+    This uses the same method as rdflib's isomorphic function to generate a cryptographic hash of a given graph.
+    This can be used to calcualte graph isomorphism across time without needing to save the entire graph.
+
+    :param graph: graph to hash
+    :type graph: graph
+
+    :return: integer hash
+    :rtype: int
+    """
+    # Copy graph to memory (improved performance if graph is backed by a DB store)
+    graph_prime = copy_graph(graph)
```

Comment:
The one question I have is if I should just create a new graph and add the db graph to it instead of calling copy graph, I think the former may be more performant and I'm not sure I need the blank node preservation in this usecase.

Reply:
Copying the graph is definitely slower than I want it to be. This is a property of the underlying graph store, so if you are adding a bunch of tuples and not doing some sort of copy-on-write thing then I'm not sure how much faster it could get

Reply:
What's the performance difference if you do this on a copy vs doing it on the original graph?

Reply:
<img width="964" alt="image" src="https://github.com/NREL/BuildingMOTIF/assets/4641913/f8d0734a-0c05-4b61-9f4c-c665647317a3">
These are the results I got with an in memory sqlite database. seems like your copy graph function is the fastest.